api/handlers/v1/resource.go

```diff
@@ -178,6 +177,33 @@ func (server APIServer) ApplyAction(ctx context.Context, request *entropyv1beta1
 	return response, nil
 }
 
+func (server APIServer) GetLog(request *entropyv1beta1.GetLogRequest, stream entropyv1beta1.ResourceService_GetLogServer) error {
+	ctx := stream.Context()
+	res, err := server.resourceService.GetResource(ctx, request.GetUrn())
+	if err != nil {
+		if errors.Is(err, store.ResourceNotFoundError) {
+			return status.Error(codes.NotFound, "could not find resource with given urn")
+		}
+		return ErrInternal
+	}
+	logChunks, err := server.moduleService.Log(ctx, res, request.GetFilter())
+	if err != nil {
+		return ErrInternal
+	}
+	for logChunk := range logChunks {
+		err := stream.Send(&entropyv1beta1.GetLogResponse{
+			Chunk: &entropyv1beta1.LogChunk{
+				Data:   logChunk.Data,
+				Labels: logChunk.Labels,
+			},
+		})
+		if err != nil {
+			return ErrInternal
+		}
```

Comment:
can it be more compact like this?
```suggestion
		if err := stream.Send(&entropyv1beta1.GetLogResponse{
			Chunk: &entropyv1beta1.LogChunk{
				Data:   logChunk.Data,
				Labels: logChunk.Labels,
			},
		}); err != nil {
			return ErrInternal
		}
```

Reply:
I would prefer the former just for the sake of clarity, I prefer to not use this when the assignment statements are not short.
We could make it short by having a function to generate the response payload.

```
func getLogResponse(logChunk domain.LogChunk) *entropyv1beta1.GetLogResponse {
	return &entropyv1beta1.GetLogResponse{
		Chunk: &entropyv1beta1.LogChunk{
			Data:   logChunk.Data,
			Labels: logChunk.Labels,
		},
	}
}
```

WDYT?

Reply:
@rohilsurana i see got it, I think it is fine to do like this
```
		err := stream.Send(&entropyv1beta1.GetLogResponse{
			Chunk: &entropyv1beta1.LogChunk{
				Data:   logChunk.Data,
				Labels: logChunk.Labels,
			},
		})
		if err != nil {
			return ErrInternal
		}
```